control-plane/pkg/apis/config/features.go

```diff
@@ -58,7 +58,8 @@ var (
 func init() {
 	DefaultTriggersConsumerGroupTemplate, _ = template.New("triggers.consumergroup.template").Parse("knative-trigger-{{ .Namespace }}-{{ .Name }}")
 	DefaultBrokersTopicTemplate, _ = template.New("brokers.topic.template").Parse("knative-broker-{{ .Namespace }}-{{ .Name }}")
-	DefaultChannelsTopicTemplate, _ = template.New("channels.topic.template").Parse("knative-channel-{{ .Namespace }}-{{ .Name }}")
+	// This will resolve to the old naming convention, to prevent errors switching over to the new topic templates approach
+	DefaultChannelsTopicTemplate, _ = template.New("channels.topic.template").Parse("knative-messaging-kafka.{{ .Namespace }}.{{ .Name }}")
```

Comment:
`knative-messaging-kafka.{{ .Namespace }}.{{ .Name }}` should be set here too: https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml#L50C3-L50C26, we don't want to change the default values (I think, that is the reason upgrade test fails, it's not a flake)

Reply:
Thanks for catching this!! I'll fix it :+1: